pkg/gardenlet/controller/shoot/state/reconciler.go

```diff
@@ -55,5 +66,59 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		return reconcile.Result{}, fmt.Errorf("error retrieving object from store: %w", err)
 	}
 
-	return reconcile.Result{}, nil
+	// if shoot got deleted or is no longer managed by this gardenlet (e.g., due to migration to another seed) then don't requeue
+	if shoot.DeletionTimestamp != nil || pointer.StringDeref(shoot.Spec.SeedName, "") != r.SeedName {
+		return reconcile.Result{}, nil
+	}
+
+	if !shootCreatedSuccessfully(shoot.Status) {
+		log.Info("Requeuing because shoot was not yet successfully created", "requeueAfter", RequeueWhenShootIsNotYetCreated)
+		return reconcile.Result{RequeueAfter: RequeueWhenShootIsNotYetCreated}, nil
```

Comment:
Instead of requeuing with a long duration, we could also adapt the update predicate to return true when the shoot just got successfully created.

Reply:
Yes sure, I was thinking about it as well, but eventually decided against. We need this `if` condition in the reconciler anyways, and since periodic backups of the `ShootState`s are not time-critical, I decided to save the extra-effort of changing the watch predicates as suggested above. It's simply less code this way, and since we jitter anyways when computing the next backups, ~`10m` should still be good enough. If you insist, I can change it, but I don't see value in it.

Reply:
I was also unsure if it was worth the effort. So agreed, let's keep it like this.